src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,16 +1,74 @@
 package wolox.training.controllers;
 
-import org.springframework.stereotype.Controller;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
 
-@Controller
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
+  @Autowired
+  private BookRepository bookRepository;
+
   @GetMapping("/greeting")
-  public String greeting(@RequestParam(name = "name", required = false, defaultValue = "World") String name, Model model) {
+  public String greeting(
+      @RequestParam(name = "name", required = false, defaultValue = "World") String name,
+      Model model) {
     model.addAttribute("name", name);
     return "greeting";
   }
-}
+
+  @GetMapping
+  public Iterable findAll() {
+    return bookRepository.findAll();
+  }
+
+  @PostMapping("/create")
+  @ResponseStatus(HttpStatus.CREATED)
+  public Book create(@RequestBody Book book) {
+    return bookRepository.save(book);
+  }
+
+  @GetMapping("/details/{id}")
+  @ResponseStatus(HttpStatus.ACCEPTED)
+  public Book findOne(@PathVariable int id) throws BookNotFoundException {
+    Optional<Book> book = bookRepository.findById(id);
```

Comment:
You can change the logic here by leveraging the fact that you are getting an Optional from the repository.
Because of that, you can do: ```bookRepository.findById(id).orElseThrow(new BookNotFoundException("Book not found"));```
Also, if you do the BookNotFoundException set that text automatically (given a BookNotFoundException will probably be always  because the book wasn't found) in the default constructor instead, you can do
`BookNotFoundException::new` instead of `new BookNotFoundException("Book not found")`